circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -38,10 +40,17 @@
   private final static Logger LOG = LoggerFactory.getLogger(JceksAmazonS3ClientFactory.class);
 
   private final Security security;
+  private final HiveConf conf;
 
   @Autowired
+  public JceksAmazonS3ClientFactory(Security security, HiveConf replicaHiveConf) {
```

Comment:
Why do we need the hive configuration passed in now?

Reply:
The role is added to the configuration, and thats then given to the credential provider chain. It's needed so that `AssumeRoleCredentialProvider` can be used